tests/playwright/src/sso-extension.spec.ts

```diff
@@ -0,0 +1,160 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { Page, Locator, expect as playExpect } from '@playwright/test';
+import { AuthenticationPage, NavigationBar, PodmanDesktopRunner, RunnerTestContext, SSOExtensionPage, SettingsExtensionsPage, WelcomePage } from '@podman-desktop/tests-playwright';
+import { afterAll, beforeAll, describe, test, beforeEach } from 'vitest';
+
+let pdRunner: PodmanDesktopRunner;
+let page: Page;
+let navBar: NavigationBar;
+let extensionInstalled = false;
+const imageName = 'ghcr.io/redhat-developer/podman-desktop-redhat-account-ext:latest';
+const extensionName = 'Red Hat Authentication';
+const extensionLabelName = 'redhat-authentication';
+const authProviderName = 'Red Hat SSO';
+
+beforeEach<RunnerTestContext>(async ctx => {
+  ctx.pdRunner = pdRunner;
+});
+
+beforeAll(async () => {
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
+  pdRunner.setVideoAndTraceName('sso-e2e');
+
+  const welcomePage = new WelcomePage(page);
+  await welcomePage.handleWelcomePage(true);
+  navBar = new NavigationBar(page);
+});
+
+afterAll(async () => {
+  await pdRunner.close();
+});
+
+describe('Red Hat Authentication extension verification', async () => {
+
+  test('Go to settings and check if extension is already installed', async () => {
+    const settingsBar = await navBar.openSettings();
+    const extensions = await settingsBar.getCurrentExtensions();
+    if (await extensionExists(extensions, extensionName)) {
+      extensionInstalled = true;
+    }
+  });
+
+  test.runIf(extensionInstalled)(
+    'Uninstalled previous version of sso extension',
+    async () => {
+      await removeExtension(extensionName);
+    },
+    60000,
+  );
+
+  test('Extension can be installed using OCI image', async () => {
+    const settingsBar = await navBar.openSettings();
+    const settingsExtensionPage = await settingsBar.openTabPage(SettingsExtensionsPage);
+    await playExpect(settingsExtensionPage.heading).toBeVisible();
+    await settingsExtensionPage.installExtensionFromOCIImage(imageName);
+  }, 200000);
+
+  test('Extension record is added under installed extensions', async () => {
+    const settingsBar = await navBar.openSettings();
+    const settingsExtensionPage = await settingsBar.openTabPage(SettingsExtensionsPage);
+    const extension = settingsExtensionPage.getExtensionRowFromTable(extensionLabelName);
+    await playExpect(extension).toBeVisible();
+    await extension.scrollIntoViewIfNeeded();
+    const extensionDetails = extension.getByRole('cell', { name: 'Extension Details' });
+    const details = extensionDetails.getByLabel('Connection Status Label');
```

Comment:
I don't think this and the previous line are needed, the Connection Status Label is exposed via the status property from the ExtensionPage class which is inhereted in the SSO extension class.

Reply:
This is SettingsExtensionsPage (inherits from SettingsPage) which is different from SSOExtensionPage.

Reply:
As far as I can tell the SSOExtensionPage inherits from ExtensionPage which also inherits from SEttingsPage

Reply:
![image](https://github.com/redhat-developer/podman-desktop-redhat-account-ext/assets/106525396/04d23b1e-75c4-4906-a3c4-3ff6045ccfe2)

The properties are already available in ExtensionPage which SSOExtensionPAge inherits.

Reply:
ha looking at that screenshot I realized there's a bug in the enableExtension() method, we're calling the disableButton instead of enableButton.

Reply:
oh, I just put that in, without actually using it. Since it is different page :)

Reply:
SettingsExtensionPage:
![Screenshot_20240411_151616](https://github.com/redhat-developer/podman-desktop-redhat-account-ext/assets/19164299/ba85306a-4f1e-4dc7-98b8-1b4d6f381bd8)

vs.
SSOExtensionPage:
![Screenshot_20240411_151721](https://github.com/redhat-developer/podman-desktop-redhat-account-ext/assets/19164299/ec5b6532-4901-4dc7-b0dc-b2a5a63b13d7)

Reply:
ah I see, you're not actually going to the specific extension page, you're just checking it in the screen where you can also install them via OCI

Reply:
Yes. It took me quite a lot to get actually oriented between the same paradigm of SEttings's extension and Extension's page.